### PR TITLE
Bug fix: path parameters are not solved for the pattern /pet/{petID}/image/*

### LINKED ISSUE #2509: Path Parameters are not populated for some cases with wildcard templates
### Description:
Path Parameters are not populated for a template like /pet/{petID/somestring/* when the rawURL is /pet/1/somestring.

In addition, following log is printed.

```
PathTemplate and RawPath is mismatched.
```

### Affected Product Version:
choreo-connect-1.0.0-beta

### Environment details (with versions):
- OS:
- Client:
- Env (Docker/K8s):

---

### PR DESCRIPTION
**Pull request by VirajSalaka**
### Issues

Fixes #2509 

### Automation tests
 - Unit tests added: Yes
 - Integration tests added: No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — APPROVED** [review]

File: enforcer-parent/commons/src/test/java/org/wso2/choreo/connect/enforcer/commons/model/RequestContextTest.java [hunk 1]
```diff
@@ -101,6 +105,9 @@ private void testPathParamValues(String rawPath, String basePath, String pathTem
         builder.matchedAPI(new APIConfig.Builder("Petstore").basePath(basePath).build());
         builder.pathTemplate(pathTemplate);
         RequestContext requestContext = builder.build();
+        if (pathParamName == null) {
+            return;
+        }
```
Comment (praminda) [inline enforcer-parent/commons/src/test/java/org/wso2/choreo/connect/enforcer/commons/model/RequestContextTest.java:110]: i dont think its safe to have this. shall we remove this as we are not having a null case in our tests?
```suggestion
```

File: enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/model/ParameterResolver.java [hunk 2]
```diff
@@ -63,7 +65,7 @@ public ParameterResolver(final String parameterTemplate) {
         final Matcher matcher = pattern.matcher(formattedURI);
         if (!matcher.matches()) {
             // Unlikely to occur as this pair is already matched within router.
-            logger.error("PathTemplate and RawPath is mismatched.");
+            logger.debug("PathTemplate:" + pathTemplate + " and RawPath:" + uriString + " is mismatched.");
```
Comment (praminda) [inline enforcer-parent/commons/src/main/java/org/wso2/choreo/connect/enforcer/commons/model/ParameterResolver.java:68]: shall we use templated log format?
```suggestion
            logger.debug("PathTemplate: {}  and RawPath: {} is mismatched.", pathTemplate, uriString);
```

**Review by praminda — APPROVED** [review]

File: enforcer-parent/commons/src/test/java/org/wso2/choreo/connect/enforcer/commons/model/RequestContextTest.java [hunk 3]
```diff
@@ -72,8 +72,14 @@ public void testPathParametersWithNumbers() {
...
         testPathParamValues("/v2/pet/12/2/random/random2", "/v2", "/pet/{petId}/{imageId}/*",
                 "petId", "12");
+        testPathParamValues("/v2/pet/12/2/random/random2/random3", "/v2", "/pet/{petId}/{imageId}/*",
+                "petId", "12");
         testPathParamValues("/v2/pet/12/image/2/random/random2", "/v2", "/pet/{petId}/image/{imageId}/*",
                 "imageId", "2");
+        testPathParamValues("/v2/pet/12/2/image", "/v2", "/pet/{petId}/{imageId}/image/*",
+                "petId", "12");
+        testPathParamValues("/v2/pet/12/2/image123", "/v2", "/pet/{petId}/{imageId}/image/*",
+                null, null);
```
Comment (AmaliMatharaarachchi) [inline enforcer-parent/commons/src/test/java/org/wso2/choreo/connect/enforcer/commons/model/RequestContextTest.java:82]: Are we allowing the following? shall we add following cases as well if not added already?
```suggestion
        testPathParamValues("/v2/pet/12/2/image", "/v2", "/pet/{imageId}/image",
        "imageId", "12/2"); 
        testPathParamValues("/v2/pet/12/image/*", "/v2", "/pet/{imageId}/image/*",
        "imageId", "12"); 
```

Comment (VirajSalaka) [reply enforcer-parent/commons/src/test/java/org/wso2/choreo/connect/enforcer/commons/model/RequestContextTest.java]: added those test cases.
except 
`testPathParamValues("/v2/pet/12/2/image", "/v2", "/pet/{imageId}/image","imageId", "12/2");` - invalid
`testPathParamValues("/v2/pet/12/image/*", "/v2", "/pet/{imageId}/image/*", "imageId", "12"); ` - already covered.

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] dev-deployment : 20211215.32

**Comment by choreo-cicd** [comment]
[succeeded] dev-deployment : 20211215.43

**Pull request merged**